components/usage/pkg/stripe/stripe.go

```diff
@@ -152,18 +152,6 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
 		return nil, fmt.Errorf("failed to register usage for customer %q on subscription item %s", customer.Name, subscriptionItemId)
 	}
 
-	invoice, err := c.GetUpcomingInvoice(ctx, customer.ID)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find upcoming invoice for customer %s: %w", customer.ID, err)
-	}
-
-	_, err = c.UpdateInvoiceMetadata(ctx, invoice.ID, map[string]string{
-		ReportIDMetadataKey: summary.ReportID,
-	})
-	if err != nil {
-		return nil, fmt.Errorf("failed to udpate invoice %s metadata with report ID: %w", invoice.ID, err)
-	}
-
```

Comment:
Dropping this part as 
1. It already doesn't work as we can't update an UpcomingInvoice in Stripe with metadata
2. We're no longer planning to use the metadata for sharing the report